api/PayrollProcessor.Data.Persistence/Infrastructure/Clients/CosmosClientExtensions.cs

```diff
@@ -30,4 +30,9 @@ public static class CosmosClientExtensions
                 PartitionKey = new PartitionKey(partitionKey.ToLowerInvariant())
             });
     }
+
+    public static class CosmosResponse
+    {
+        public static T Unwrap<T>(Response<T> response) => response.Resource;
```

Comment:
In the future we might want to validate the `StatusCode` property of the `Response<T>`... right now we assume everything is 👍